src/wallet/wallet2.cpp

```diff
@@ -2784,6 +2790,7 @@ void wallet2::pull_and_parse_next_blocks(uint64_t start_height, uint64_t &blocks
       }
     }
     waiter.wait(&tpool);
+    last = !blocks.empty() && cryptonote::get_block_height(parsed_blocks.back().block) + 1 == blocks_start_height + parsed_blocks.size();
```

Comment:
Isn't this always true (given that `blocks` is not empty)?
```cryptonote::get_block_height(parsed_blocks.back().block) + 1 == blocks_start_height + parsed_blocks.size()```

Reply:
It should be if the sender sends correct data.

Reply:
But will `refresh` handle fetching 2500 blocks (it needs 3 `get_blocks` calls)?
Seems the first `pull_and_parse_next_blocks` invocation will set `last` to `true`.

Reply:
Pretty sure I tried that. I'll check.

Reply:
Looks like I lost some patch, I was using the returned chain height. Fixing, you are right,